### ISSUE TITLE
No diary entry shortly after midnight

### ISSUE THREAD

**Issue by DerVogel2020** [corona-warn-app/cwa-app-android#2740]
## Avoid duplicates
* [x ] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [x ] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [x ] Bug is not already reported in another issue

## Technical details

- Device name: Samsung Galaxy A5 2017
- Android version: 8
- App version: 1.15.2

## Describe the bug

When opening the diary shortly after midnight, there is no entry for today. See screenshot.
![20210405_073017.jpg](https://user-images.githubusercontent.com/65366950/113542219-19a11000-95e4-11eb-8d41-ebe8d9491c4f.jpg)
At about 1:10, there still was no entry.
At about 7:00, the entry appeared.

## Steps to reproduce the issue

Open the diary shortly after midnight. 

## Expected behaviour

Show the diary entry foe today at 0:01.
## Additional context

Maybe it's because of the time change at end of March.

JIRA: [6092](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6092)

**Comment by Ein-Tim** [comment]
@DerVogel2020 

What happens if you 

a) close the app 
b) switch to the main screen and go back to the contact diary? 

Is there still no entry?

**Comment by MikeMcC399** [comment]
@DerVogel2020 

It looks like the calendar is working on UTC time not local time (CEST = UTC+2 currently).

I enabled flight mode, disabled automatic date and time, then set the clock to tomorrow (Tuesday April 6, 2021)
00:10 then
01:10 then
02:10

It was only when I got to 2 AM that the calendar entry for Tuesday appeared. I tried Force Stopping the app each time to make sure it was not a refresh issue.

Afterwards I set the clock back again to 01:10 and the Tuesday entry disappeared.

Great catch!

**Comment by DerVogel2020** [comment]
> @DerVogel2020 
> 
> What happens if you 
> 
> a) close the app 
> b) switch to the main screen and go back to the contact diary? 
> 
> Is there still no entry? 

Tried both after manually changing date and time. 
The entry didn't appear.

**Comment by DerVogel2020** [comment]
> @DerVogel2020 
> 
> It looks like the calendar is working on UTC time not local time (CEST = UTC+2 currently).
> 
> I enabled flight mode, disabled automatic date and time, then set the clock to tomorrow (Tuesday April 6, 2021)
> 00:10 then
> 01:10 then
> 02:10
> 
> It was only when I got to 2 AM that the calendar entry for Tuesday appeared. I tried Force Stopping the app each time to make sure it was not a refresh issue.
> 
> Afterwards I set the clock back again to 01:10 and the Tuesday entry disappeared.
> 
> Great catch!

Just set the date and time to 2am and the entry appeared.
So is it a feature to change from UTC to local time or a bug?

**Comment by MikeMcC399** [comment]
@DerVogel2020 
> So is it a feature to change from UTC to local time or a bug?

The [Google API](https://developers.google.com/android/exposure-notifications/exposure-notifications-api) uses UTC times, but I would expect the user UI to translate into local time. I think it is a bug, but we'll have to wait for dev feedback to see what they say.

**Comment by harambasicluka** [comment]
I agree this shouldn't happen, I would also classify it as bug. I'll let you know when we fix it :)

Thanks,
Luka

**Comment by harambasicluka** [comment]
@DerVogel2020 the bug will be fixed with https://github.com/corona-warn-app/cwa-app-android/pull/2835

**Comment by Ein-Tim** [comment]
Version 2.1 has been released today, can this be closed?

**Comment by DerVogel2020** [comment]
I didn't receive the update yet, but will check afterwards.

**Comment by MikeMcC399** [comment]
@Ein-Tim 
> Version 2.1 has been released today, can this be closed?

Please factor in that Android users may have to wait 48 hours until being offered a new release. I also don't have the new release yet and I like to be able to verify any fixes that I have been involved in using the released version (instead of a version built by myself with Android Studio). So please be a little patient before asking if Android issues can be closed - maybe asking after one week following a release if there is no movement in an issue would be a good rule of thumb.

**Comment by Ein-Tim** [comment]
@MikeMcC399 

Okay, I'll try to be more patient here (: 

Thank you for your feedback 👍

**Comment by MikeMcC399** [comment]
@DerVogel2020 

Now that I managed to access the new 2.1 release from Google Play Store I went through the repro steps from https://github.com/corona-warn-app/cwa-app-android/issues/2740#issuecomment-813443209.

I enabled flight mode, disabled automatic date and time, then set the clock to tomorrow (Tuesday May 4, 2021) with time 00:10.

The calendar entry for Tuesday appeared with the clock set to 00:10. As far as I can see, the issue is fixed.

You might like to check this for yourself. Great that you spotted this issue and reported it!
